app/models/house.rb

```diff
@@ -3,10 +3,27 @@ class House < ApplicationRecord
   has_many :home_owners_associations, through: :hood
   has_one :house_metadatum, dependent: :destroy
 
+  delegate :zpid, to: :house_metadatum, allow_nil: true
+
+  after_initialize :set_price_history
+
   PRICE_HISTORY_DATE_FORMAT = "%Y-%m-%d".freeze
 
   def valuation_on(date)
     normalized_date = date.strftime(PRICE_HISTORY_DATE_FORMAT)
     price_history.dig(normalized_date).to_f
   end
+
+  def add_valuation(date, price)
```

Comment:
This is the meat and potatoes of this PR. No other things that use a house should know what `price_history`'s shape is. We should provide an interface to insert and retrieve from it.

Reply:
I like it